src/Fantomas/AstTransformer.fs

```diff
@@ -9,41 +9,20 @@ open Fantomas
 type Id = { Ident: string; Range: Range }
 
 module Helpers =
-    let i (id: Ident) : Id =
-        { Ident = id.idText
-          Range = id.idRange }
-
-    let li (id: LongIdent) = id |> List.map i
-
-    let lid (id: LongIdentWithDots) = li id.Lid
     let mkNode (t: FsAstType) (r: range) = TriviaNodeAssigner(MainNode(t), r)
 
 module private Ast =
     open Helpers
 
     let rec visitSynModuleOrNamespace (modOrNs: SynModuleOrNamespace) : TriviaNodeAssigner list =
         match modOrNs with
-        | SynModuleOrNamespace (longIdent, _, synModuleOrNamespaceKind, decls, _, attrs, _, range) ->
-            let collectIdents (idents: LongIdent) =
-                idents
-                |> List.map (fun ident -> mkNode Ident_ ident.idRange)
-
-            let typeName =
-                match synModuleOrNamespaceKind with
-                | SynModuleOrNamespaceKind.AnonModule -> SynModuleOrNamespace_AnonModule
-                | SynModuleOrNamespaceKind.NamedModule -> SynModuleOrNamespace_NamedModule
-                | SynModuleOrNamespaceKind.DeclaredNamespace -> SynModuleOrNamespace_DeclaredNamespace
-                | SynModuleOrNamespaceKind.GlobalNamespace -> SynModuleOrNamespace_GlobalNamespace
-
-            [
-              // LongIdent inside Namespace is being processed as children.
-              if typeName <> SynModuleOrNamespace_DeclaredNamespace then
-                  mkNode typeName range
-              yield!
-                  if synModuleOrNamespaceKind = SynModuleOrNamespaceKind.DeclaredNamespace then
-                      collectIdents longIdent
-                  else
-                      []
+        | SynModuleOrNamespace (longIdent, _, kind, decls, _, attrs, _, range) ->
+            let longIdentNodes =
+                match kind, decls with
+                | SynModuleOrNamespaceKind.AnonModule, _ :: _ -> []
```

Comment:
If an anonymous module has any declarations, those will be better trivia candidates later.